packages/push/src/PushRegistration.ts

```diff
@@ -61,14 +64,28 @@ export class PushRegistration {
       return Promise.reject(new Error("UPS registration: Platform is configured." +
         "Please add UPS variant and generate mobile - config.json again"));
     }
+
     const variantId = platformConfig.variantId || platformConfig.variantID;
-    const variantSecret = platformConfig.variantSecret;
     if (!variantId) {
       return Promise.reject(new Error("UPS registration: variantId is not defined."));
     }
+
+    const variantSecret = platformConfig.variantSecret;
     if (!variantSecret) {
       return Promise.reject(new Error("UPS registration: variantSecret is not defined."));
     }
+
+    this._objectInstance = window.PushNotification.init(
```

Comment:
This looks like registration plugin getting direct connection to cordova plugin. 
Idea was that no matter what platform it will be it will be just doing the registration.
Just mentioning.

Reply:
Hrm, so... how to handle FCM topic subscriptions?

Reply:
I'm just saying that this package was purely to do UPS registrations. 
FCM and other stuff is Phonegap Push Plugin related so bringing this here ties this entire package to PhoneGap push. 

It is nice to get helpers for this stuff, so maybe we can add docs and showcase integration as well. 

The entire helper is really basic and it is a calling subscribe method so maybe showcase should have this and we should document that.

Feel free to ignore me here.

Reply:
Yes. Ignore me here. I have already approved changes.
Let's fix the build to do verification

Reply:
I got your point and totally agree with you. The only point is without this the app will not be able to receiver message from UPS. Only if the message is sending directly to the device using an alias.

TL:DR when you send a message on UPS without any criteria it will use FCM topic to delivery a message for all devices registered on the topic (with the same name/id of the variant) same for the category.

If you understood you correctly, what you are recommending is to move that code/feature to the `push-cordova` like a helper method and document it's needed to be fired if the app is running on Android devices. Am I correct?

Reply:
I think is ok to have it this way. Alternatively we can have a strategy that will be default assume and use window.PushNotification.register is default one (if exist) and user can pass something to constructor. That way we provide the same functionality but people could still use let's say old plugin or anything they will need to use (like web) etc.

Reply:
@wtrocki is right about the original intention of this library which was to provide UPS registration only and not be coupled to the PhoneGap plugins. This would allow this library to be portable to the browser and to other platforms like React Native. However I also get @wei-lee and @danielpassos's point about making it easier to perform the actual registration with firebase.

One of the problems with the current approach in this PR is that the `register` function will now fail (silently) in all platforms except on a device. I think this approach is not optimal.

Instead I think we should provide the PhoneGap push registration in a separate function that is not called immediately when a user calls `register()`. 

I think the usage in a user's app should be something like this.

```js
const push = new PushRegistration(config) // the name of this class kinda sucks as well 😄
await push.registerUPS(token, alias, categories) // register with UPS only

try {
  await push.registerFirebaseTopics(categories) // perform the extra phonegap specific registration
} catch(e) {
  // handle error. Could be because wrong platform or could be a legit error.
}